tools/ReleaseTool/Common.cs

```diff
@@ -40,5 +44,21 @@ public static string AppendDirectorySeparator(string originalPath)
 
             return originalPath + directorySeparator;
         }
+
+        public class WorkingDirectoryScope : IDisposable
+        {
+            private readonly string oldWorkingDirectory;
+            
+            public WorkingDirectoryScope(string newWorkingDirectory)
+            {
+                oldWorkingDirectory = Environment.CurrentDirectory;
+                Environment.CurrentDirectory = newWorkingDirectory;
```

Comment:
The setter for `CurrentDirectory` can throw a number of exceptions - how are you handling this?

Reply:
Hmmm I'm happy the exceptions to propagate out here in the constructor since it means you've given this garbage data (and I'd otherwise do the exact same checks that the setter is doing).

However, there's also the chance that it throws in the `Dispose()` (which would be odd as it meant the path from the original directory is either: null or empty, an invalid path, perms err). This could be thrown if the user removes the original directory while inside the scope, but not sure what the graceful thing to do is though.

We really shouldn't throw exceptions in disposes though..

Reply:
Discussed offline - in this case, it may be best to let the exceptions bubble up since if these options fail its a full on failure case